modules/apps/fragment/fragment-impl/src/main/java/com/liferay/fragment/internal/processor/PortletRegistryImpl.java

```diff
@@ -127,12 +127,22 @@ public List<String> getFragmentEntryLinkPortletIds(
 
 	@Override
 	public List<String> getPortletAliases() {
-		return new ArrayList<>(_portletNames.keySet());
+		return new ArrayList<>(_aliasPortletNames.keySet());
 	}
 
 	@Override
 	public String getPortletName(String alias) {
-		return _portletNames.get(alias);
+		return _aliasPortletNames.get(alias);
+	}
+
+	@Override
+	public void registerAlias(String alias, String portletName) {
+		_aliasPortletNames.put(alias, portletName);
+	}
+
+	@Override
+	public void unregisterAlias(String alias, String portletName) {
+		_aliasPortletNames.remove(alias, portletName);
```

Comment:
@W2jx Why you have to remove by both key and value?

Reply:
Hi @tinatian,  here I am just following the original logic. It is possible to remove it only by key, because in our code, alias is unique.